### ISSUE TITLE
feat(core): extract export_onnx → ExportResult (W&B-free)

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#87]
## Goal

Extract the pure ONNX export logic out of `core.registry.promote_to_registry` into a new W&B-free function `core.export_onnx(...) -> ExportResult` in `radiologist-core`. Given a checkpoint path and export metadata, it loads the Lightning module, resolves the CAM target layer, and writes two ONNX files — a deterministic export and an MC-Dropout export — returning an `ExportResult` (imported from `radiologist-registry`) with both local paths plus the run id, input shape, and class list. This establishes the `core -> registry` dependency direction. Depends on #85.

## Context

Export logic to extract from `radiologist-core/src/radiologist/core/registry/promote.py` (steps 4–8 of `promote_to_registry`), with all W&B steps (1–3, 9–10) removed:

```python
# 4. LModule.load_from_checkpoint(ckpt_path, map_location="cpu", weights_only=False)
# 5. resolve cam_target_layer via _resolve_layer(net, dot_path)
# 6. Build base_meta dict
# 7. Deterministic ONNX:
#    - net.train(mode=False); wrap in _CamWrapper; wrapper.train(mode=False)
#    - torch.onnx.export(..., output_names=["logits", "feature_maps"])
#    - embed det_meta into ONNX metadata_props
# 8. MC-Dropout ONNX:
#    - net.train(mode=False); set all Dropout submodules to .train()
#    - torch.onnx.export(..., training=TrainingMode.PRESERVE, do_constant_folding=False)
#    - embed mcd_meta into ONNX metadata_props
```

### Proposed signature (new `core.export_onnx`)

```python
from typing import List, Tuple

from radiologist.registry import ExportResult


def export_onnx(
    ckpt_path: str,
    run_id: str,
    input_shape: Tuple[int, ...],
    classes: List[str],
    cam_target_layer: str,
    out_dir: str,
    opset: int = 18,
) -> ExportResult:
    """Load a Lightning checkpoint and write deterministic + MC-Dropout ONNX files.

    Returns an ExportResult with both local ONNX paths plus run_id, input_shape,
    and classes (no W&B interaction).
    """
```

`run_id`, `input_shape`, and `classes` are passed through onto the returned `ExportResult`. Determining `run_id`/`precision` from the source run is now the caller's responsibility (issue #4/#6), not this function's — it no longer touches W&B.

### `ExportResult` (imported from `radiologist.registry`)

```python
@dataclass(frozen=True)
class ExportResult:
    det_path: str
    mcd_path: str
    run_id: str
    input_shape: Tuple[int, ...]
    classes: List[str]
```

### Gotchas

- `model.eval()` is forbidden (sandbox false-positive) — use `model.train(mode=False)` everywhere, including the wrapper and the per-`Dropout`-submodule re-enable in the MC-Dropout branch.
- `weights_only=False` is required on `load_from_checkpoint` (Lightning checkpoints embed Python objects; safe as the artifact is our own).
- Embed metadata into `onnx.ModelProto.metadata_props` after export.
- `core` already contains `_resolve_layer`, `_CamWrapper`, and the metadata-embedding helpers used by `promote.py` — reuse them; do not reimplement.
- Python 3.10 typing: `Optional[X]`, `Tuple[X, ...]`, `List[X]` from `typing`.

### Create

- `radiologist-core/src/radiologist/core/registry/export.py` — `export_onnx(...) -> ExportResult` plus any private export helpers it needs (reusing existing `_resolve_layer` / `_CamWrapper` from the existing registry module).
- `radiologist-core/tests/test_onnx_export.py` — behavior tests on the two written ONNX files.

### Modify

- `radiologist-core/pyproject.toml` — add `radiologist-registry>=0.1.0` to `dependencies` and `radiologist-registry = { workspace = true }` under `[tool.uv.sources]` (mirror existing workspace deps).
- `radiologist-core/src/radiologist/core/registry/__init__.py` — export `export_onnx` and add it to `__all__`.

## Acceptance criteria

- [ ] Given a valid checkpoint path and export metadata, `export_onnx(...)` writes two ONNX files under `out_dir` and returns an `ExportResult` whose `det_path` and `mcd_path` point at existing files on disk.
- [ ] The returned `ExportResult` carries the `run_id`, `input_shape`, and `classes` passed in, unchanged.
- [ ] The deterministic ONNX exposes output names `logits` and `feature_maps` and carries the deterministic metadata in its `metadata_props`.
- [ ] The MC-Dropout ONNX is exported with dropout layers preserved in training mode and carries the MC-Dropout metadata in its `metadata_props`.
- [ ] `export_onnx` performs no W&B calls (importing it does not require `wandb`).
- [ ] mypy clean; pytest green.

## Out of scope

- Uploading / linking artifacts to a W&B collection (issue #88).
- Removing `promote_to_registry` / `pull_checkpoint` from `core.__all__` (issue #90).
- The Predictor refactor (separate epic).
